pkg/core/store/badger/extension.go

```diff
@@ -0,0 +1,340 @@
+package badger
+
+import (
+	"context"
+	"time"
+
+	"github.com/fuseml/fuseml-core/pkg/domain"
+	"github.com/timshannon/badgerhold/v3"
+)
+
+// ExtensionStore is a wrapper around a badgerhold.Store that implements the domain.ExtensionStore interface.
+type ExtensionStore struct {
+	store *badgerhold.Store
+}
+
+// NewExtensionStore creates a new ExtensionStore.
+func NewExtensionStore(store *badgerhold.Store) *ExtensionStore {
+	return &ExtensionStore{store: store}
+}
+
+// AddExtension adds a new extension to the store.
+func (es *ExtensionStore) AddExtension(ctx context.Context, extension *domain.Extension) (*domain.Extension, error) {
+	extension.EnsureID(ctx, es)
+	extension.SetCreated(ctx)
+
+	err := es.store.Insert(extension.ID, extension)
+	if err != nil {
+		return nil, domain.NewErrExtensionExists(extension.ID)
+	}
+	return extension, nil
+}
+
+// GetExtension retrieves an extension by its ID.
+func (es *ExtensionStore) GetExtension(ctx context.Context, extensionID string) (*domain.Extension, error) {
+	extension := &domain.Extension{}
+	err := es.store.Get(extensionID, extension)
+	if err != nil {
+		return nil, domain.NewErrExtensionNotFound(extensionID)
+	}
+	return extension, nil
+}
+
+// ListExtensions retrieves all stored extensions.
+func (es *ExtensionStore) ListExtensions(ctx context.Context, query *domain.ExtensionQuery) (result []*domain.Extension) {
+	result = []*domain.Extension{}
+
+	// TODO: Replace with a badgerhold query.
+	if query != nil {
+		if query.ExtensionID != "" {
+			fullExtension, err := es.GetExtension(ctx, query.ExtensionID)
+			if err == nil {
+				matchingExtension := fullExtension.GetExtensionIfMatch(query)
+				if matchingExtension != nil {
+					result = append(result, matchingExtension)
+				}
+			}
```

Comment:
Why is the potential error ignored here?

Reply:
`GetExtension` returns an error only if the extension cannot be found and in that case this functions will return `result` (empty list). So, there is no need to check for the error as it is expected when there is no extension with the specified ID.

Reply:
Well, the question is, if this is correct behavior (returning all items when query does not match anything)

Reply:
Well in the case that it fails "all items" will be an empty list, so I would say yes.

Reply:
OK, right, empty list, not all items. The question remains the same.

But I guess that's how we do it with other resources too.

Reply:
yes, I think that we agreed on that.